rllib/policy/sample_batch.py

```diff
@@ -352,36 +352,34 @@ def split_by_episode(self) -> List["SampleBatch"]:
             [{"a": [1, 2], "eps_id": [0, 0]}, {"a": [3], "eps_id": [1]}]
         """
 
-        # No eps_id in data -> Make sure there are no "dones" in the middle
-        # and add eps_id automatically.
-        if SampleBatch.EPS_ID not in self:
-            # TODO: (sven) Shouldn't we rather split by DONEs then and not
-            #  add fake eps-ids (0s) at all?
-            if SampleBatch.DONES in self:
-                assert not any(self[SampleBatch.DONES][:-1])
-            self[SampleBatch.EPS_ID] = np.repeat(0, self.count)
+        slices = []
```

Comment:
Could we add more comments here as to why we would want to do this?

Also, can we add a test to `test_sample_batch.py` to check this?